File: src/717_1BitAnd2BitCharacters/717.scala
object Solution {
    def isOneBitCharacter(bits: Array[Int]): Boolean = bits match {
            case Array(0)    => true 
            case Array(1, 0) => false 
            case Array(1, 1) => false 
            case Array(x, _*) => if x == 0 
                then isOneBitCharacter(bits.tail)
                else isOneBitCharacter(bits.tail.tail)
        }
}

File: src/717_1BitAnd2BitCharacters/717.go
func isOneBitCharacter(bits []int) bool {
    for i := 0; i < len(bits); {
        if bits[i] == 1 {
            if i + 2 >= len(bits) { return false }
            i += 2
        } else {
            i++
        }
    }

    return true 
}